src/main/java/org/cryptomator/fusecloudaccess/CloudAccessFS.java

```diff
@@ -74,17 +83,32 @@ public int getattr(String path, FileStat stat) {
 	@Override
 	public int readdir(String path, Pointer buf, FuseFillDir filler, long offset, FuseFileInfo fi) {
 		var returnCode = provider.listExhaustively(Path.of(path)).thenApply(itemList -> {
-			filler.apply(buf, ".", null, 0);
-			filler.apply(buf, "..", null, 0);
+			//if these already return "Buffer full" something 's definitively wrong but still
+			if (filler.apply(buf, ".", null, 0) != 0
+					|| filler.apply(buf, "..", null, 0) != 0) {
+				return -ErrorCodes.ENOMEM();
+			}
+
```

Comment:
needs refactoring anyway (for pagination support...)

Reply:
I think this can be resolved, since pagination support is now done in another branch.